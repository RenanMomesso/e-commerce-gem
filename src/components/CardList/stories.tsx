import { Story, Meta } from '@storybook/react'
import PaymentOptionsMock from 'components/PaymentOptions/mock'
import CardList from '.'

export default {
  title: 'Profile/CardList',
  component: CardList,
  args: {
    cards: PaymentOptionsMock
  }
} as Meta

export const Default: Story = (args) => (
  <div style={{ maxWidth: 850, margin: 'auto' }}>
    <CardList {...args} />
  </div>
)
