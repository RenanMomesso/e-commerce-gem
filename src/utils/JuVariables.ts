export const JuJoiasVariables = {
  ecommerceName: 'JulyanaEcommerceJoias'
}
