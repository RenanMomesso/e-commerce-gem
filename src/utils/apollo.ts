import {
  ApolloClient,
  HttpLink,
  InMemoryCache,
  NormalizedCacheObject
} from '@apollo/client'
import { concatPagination } from '@apollo/client/utilities'
import { useMemo } from 'react'

let apollotClient: ApolloClient<NormalizedCacheObject | null>

function createApolloClient() {
  return new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: new HttpLink({ uri: 'http://localhost:1337/graphql' }),
    cache: new InMemoryCache({
      typePolicies: {
        Query: {
          fields: {
            products: concatPagination(['where', 'sort'])
          }
        }
      }
    })
  })
}

export function initializeApollo(initialState = null) {
  //verifica se já existe uma instância, para ñ criar outra
  const apolloClientGlobal = apollotClient ?? createApolloClient()

  //recuperando dados do cache
  if (initialState) {
    apolloClientGlobal.cache.restore(initialState)
  }

  //sempre inicializando no SSR com cache limpo
  if (typeof window === 'undefined') return apolloClientGlobal
  apollotClient = apollotClient ?? apolloClientGlobal

  return apollotClient
}

export function useApollo(initialState = null) {
  const store = useMemo(() => initializeApollo(initialState), [initialState])
  return store
}
