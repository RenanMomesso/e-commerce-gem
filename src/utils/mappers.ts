import {
  QueryHome_banners,
  QueryHome_newProducts,
  QueryHome_sections_popularProducts_highlight
} from 'graphql/generated/QueryHome'

export const bannersMap = (banners: QueryHome_banners[]) => {
  return banners.map((banner) => ({
    img: `http://localhost:1337${banner.image?.url}`,
    title: banner.title,
    subtitle: banner.subtitle,
    buttonLabel: banner.button?.label,
    buttonLink: banner.button?.link,
    ...(banner.ribbon && {
      ribbon: banner.ribbon?.text || null,
      ribbonColor: banner.ribbon?.color || null,
      ribbonSizes: banner.ribbon?.size || null
    })
  }))
}

export const productsMapper = (
  product: QueryHome_newProducts[] | null | undefined
) => {
  return (
    product &&
    product.map((productItem) => ({
      title: productItem.name,
      slug: productItem.slug,
      category: productItem.category,
      img: `http://localhost:1337${productItem.cover?.url}`,
      price: productItem.price
    }))
  )
}

export const highlightMapper = (
  highlight: QueryHome_sections_popularProducts_highlight | null | undefined
) => {
  return (
    highlight && {
      title: highlight?.title,
      subtitle: highlight?.subtitle,
      buttonLink: highlight?.buttonLink,
      buttonLabel: highlight?.buttonLabel,
      backgroundImage: `http://localhost:1337${highlight?.background?.url}`,
      alignment: highlight?.alignment,
      floatImage: highlight.floatImage
    }
  )
}
