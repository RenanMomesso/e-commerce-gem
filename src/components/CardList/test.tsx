import { screen } from '@testing-library/react'
import PaymentOptionsMock from 'components/PaymentOptions/mock'
import { renderWithTheme } from 'utils/helper'

import CardList from '.'

describe('<CardList />', () => {
  it('should render the heading', () => {
    renderWithTheme(<CardList cards={PaymentOptionsMock} />)

    expect(
      screen.getByRole('heading', { name: /my cards/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /mastercard/i })).toHaveAttribute(
      'src',
      'mastercard'
    )
    expect(screen.getByText(/3212/i)).toBeInTheDocument()
    expect(screen.getByText(/4326/i)).toBeInTheDocument()
  })
})
