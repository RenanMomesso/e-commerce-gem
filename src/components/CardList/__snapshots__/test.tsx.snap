// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CardList
  </h1>
</main>
`;
